lib/src/package_analysis/shapes.dart

```diff
@@ -0,0 +1,168 @@
+  /// top-level getter exported in this library.
+  final List<int> exportedGetters;
+
+  /// `List` of [PropertyShape.id] elements, where each one corresponds to a
+  /// top-level setter exported in this library.
+  final List<int> exportedSetters;
+
+  /// `List` of [MethodShape.id] elements, where each one corresponds to a
+  /// top-level function exported in this library.
+  final List<int> exportedFunctions;
+
+  /// `List` of [ClassShape.id] elements, where each one corresponds to a class
+  /// exported in this library.
+  final List<int> exportedClasses;
+
+  LibraryShape({
+    required this.uri,
+    required this.exportedGetters,
+    required this.exportedSetters,
+    required this.exportedFunctions,
+    required this.exportedClasses,
+  });
+
+  factory LibraryShape.fromJson(Map<String, dynamic> json) =>
+      _$LibraryShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$LibraryShapeToJson(this);
+}
+
+/// A Shape for describing a class.
+@sealed
+@JsonSerializable()
+class ClassShape extends BasicShape {
+  final List<PropertyShape> getters;
+  final List<PropertyShape> setters;
+  final List<MethodShape> methods;
+
+  ClassShape({
+    required super.id,
+    required super.name,
+    required this.getters,
+    required this.setters,
+    required this.methods,
+  });
+
+  factory ClassShape.fromJson(Map<String, dynamic> json) =>
+      _$ClassShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ClassShapeToJson(this);
+}
+
+/// A Shape for describing a class method
+@sealed
+@JsonSerializable()
+class MethodShape {
+  final String name;
+
+  MethodShape({required this.name});
+
+  factory MethodShape.fromJson(Map<String, dynamic> json) =>
+      _$MethodShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$MethodShapeToJson(this);
+}
+
+/// A Shape for describing a top-level function
+@sealed
+@JsonSerializable()
+class FunctionShape extends BasicShape {
+  FunctionShape({required super.id, required super.name});
+
+  factory FunctionShape.fromJson(Map<String, dynamic> json) =>
+      _$FunctionShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$FunctionShapeToJson(this);
+}
+
+/// A Shape for describing a getter/setter of a class property
+@sealed
+@JsonSerializable()
+class PropertyShape {
+  final String name;
+
+  PropertyShape({required this.name});
+
+  factory PropertyShape.fromJson(Map<String, dynamic> json) =>
+      _$PropertyShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$PropertyShapeToJson(this);
+}
+
+/// A Shape for describing a getter/setter of a top-level variable
+@sealed
+@JsonSerializable()
+class GlobalPropertyShape extends BasicShape {
+  GlobalPropertyShape({required super.id, required super.name});
+
+  factory GlobalPropertyShape.fromJson(Map<String, dynamic> json) =>
+      _$GlobalPropertyShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GlobalPropertyShapeToJson(this);
+}
+
+@internal
+abstract class BasicShape {
```

Comment:
```suggestion
abstract class GlobalShapeBase {
```

Reply:
because all global things have IDs, or we could call it something else.. but Base is the typical suffix.